pulsar-proxy/src/main/java/org/apache/pulsar/proxy/server/DirectProxyHandler.java

```diff
@@ -218,7 +223,7 @@ protected void handleConnected(CommandConnected connected) {
 
             state = BackendState.HandshakeCompleted;
 
-            inboundChannel.writeAndFlush(Commands.newConnected(connected.getProtocolVersion())).addListener(future -> {
+            inboundChannel.writeAndFlush(Commands.newConnected(protocolVersion)).addListener(future -> {
```

Comment:
I think this might not be correct. The intent here is to tell the client the protocol version supported by broker. By sending `protocolVersion` we send back the client's version

Reply:
Thank you @merlimat, I reverted it back.